rs/nns/handlers/root/impl/test_canisters/upgrade_test_canister.rs

```diff
@@ -1,19 +1,36 @@
 // This canister is used for testing upgrades with arguments and stable memory.
 
-use dfn_core::{api::arg_data, endpoint::over_bytes, println, stable};
+use ic_cdk::{
+    api::{
+        call::arg_data_raw,
+        stable::{stable_grow, stable_read, stable_write},
+    },
+    post_upgrade, println, query,
+};
+use std::cell::RefCell;
+
+thread_local! {
+    static ARG_LEN: RefCell<usize>  = const {RefCell::new(0) };
+}
 
 fn main() {}
 
-#[export_name = "canister_post_upgrade"]
+#[post_upgrade]
 fn post_upgrade() {
-    dfn_core::printer::hook();
-
-    let arg = arg_data();
+    let arg = arg_data_raw();
     println!("Initializing test canister with arg={:?}", arg);
-    stable::set(&arg);
+    stable_grow(1).expect("Could not grow stable memory");
```

Comment:
Why do we now need this, isn't this changing the behavior of this test canister? 

It seems we used to overwrite stable memory, and now we shift-by-one and overwrite the rest?

Reply:
Closing as non-blocking (since this is a test canister).

Reply:
The cdk did some things under the hood, and we couldn't test without making it work.